### ISSUE TITLE
quality/duplicate-dict-key misses duplicates after a comment in the dict literal

### ISSUE THREAD

**Issue by myuniquename** [atelico/gdstyle#27]
### Minimal reproduction

```gdscript
extends Node

# Case A: duplicate key with no comment in the dict literal — gdstyle catches this.
var no_comment: Dictionary = {
	Vector3i(0, 0, 0): "a",
	Vector3i(0, 0, 0): "b",
}

# Case B: same duplicate key, but a comment line sits inside the dict literal before it —
# gdstyle 0.2.3 silently stops tracking keys after the comment and misses the duplicate.
var with_comment: Dictionary = {
	# a comment inside the dict literal
	Vector3i(0, 0, 0): "a",
	Vector3i(0, 0, 0): "b",
}
```

### Command you ran

gdstyle.exe check --no-color repro.gd

### What gdstyle did

It only reports the duplicate in no_comment. Adding a comment line inside with_comment's dict literal (with or without parentheses in the comment text) makes the second duplicate go undetected.

### What you expected

quality/duplicate-dict-key reports the duplicate Vector3i(0, 0, 0) key in both no_comment and with_comment below.

### gdstyle version

gdstyle 0.2.3

### Platform

Windows (x86_64)

### Additional context

Any # comment — full-line or trailing — inside a Dictionary literal causes the rule to silently stop tracking keys for the rest of that dictionary. A genuine duplicate key that appears after such a comment is not reported.

**Comment by w4nderlust** [comment]
Nice catch, thanks for the clear repro. This is fixed in v0.2.4 (just published to GitHub Releases and crates.io), so a `cargo install gdstyle` upgrade or grabbing the new binary will pick it up.
